core/chains/legacyevm/chain.go

```diff
@@ -244,10 +244,24 @@ func newChain(ctx context.Context, cfg *evmconfig.ChainScoped, nodes []*toml.Nod
 		}
 	}
 
+	// initialize gas estimator
+	var gasEstimator gas.EvmFeeEstimator
+	if gasEstimator, err = newGasEstimator(cfg.EVM(), client, l, opts, clientsByChainID); err != nil {
+		return nil, fmt.Errorf("failed to instantiate gas estimator for chain with ID %s: %w", chainID, err)
+	}
```

Comment:
Since you're returning, there is no point in splitting the declaration from the initialization
```suggestion
	gasEstimator, err := newGasEstimator(cfg.EVM(), client, l, opts, clientsByChainID)
	if err != nil {
		return nil, fmt.Errorf("failed to instantiate gas estimator for chain with ID %s: %w", chainID, err)
	}
```

Reply:
Weird... I was getting a shadow declaration on err which is why I did this but maybe my other changes fixed it. Fixed in the latest commit